### ISSUE TITLE
Question about training an Iterseg Unet

### ISSUE THREAD

**Issue by ArtGP-79** [AbigailMcGovern/iterseg#27]
Hello!

I am writing, not to report a issue, but to ask a question about training my own affinity Unet with Iterseg.

I am trying to segment an image with the pre-trained Iterseg Unet, but it oversegments my images. Reading how to train my own Unet, I saw that I would need a 'ground truth' labelled image and the corresponding original image to do so.

My question is, how would I get the labelled ground truth in the first place? Would this require a manual segmentation?

I look forward to your reply, thank you very much for your time and help,

ArtGP-79.

**Comment by jni** [comment]
> Would this require a manual segmentation?

Yes, this is done through either manual painting *or* getting an intermediate segmentation, for example the oversegmentation from iterseg, and editing it, for example, by merging labels. (This is where the name "iterseg" comes from, although @AbigailMcGovern started with a "classical" segmentation using blob detection + watershed.) You can merge labels with napari's built-in tools for example by clicking "pick label", picking one of the labels in a cluster, then selecting "fill", and clicking on the other labels. These tools have keyboard shortcuts (you can find them by hovering), so the back and forth can be very quick. And by the way, be sure to set the "n edit dims" to 3, otherwise fill only fills a plane.

**Comment by ArtGP-79** [comment]
Hello again!

I have been working on the manual segmentation and I have re-trained the iterseg segmentation with my new 'ground truth' image. I did this by following the instructions on training iterseg ourselves with the "ground truth from ROI" section.

After re-training the segmentation algorithm, I got multiple files containing diverse documents of .tif, .txt, .pt, and .csv nature.

Trying to use the new segmentation algorithm, I did what you can see in the image:
![image](https://github.com/user-attachments/assets/e4e89198-863b-4a17-95ec-48e2411b927c)

After reaching 100% on the segmentation process, napari stops responding and crashes consistently (which is very weird because if I just segment the same image with the original iterseg without adding a network or layer reference, it works fine).

My questions in this case are the following:
- Is there any specific reason for the software to crash like this after finishing the segmentation? Should I use a more powerful computer to see if this fixes the issue or is this a file input error?
- Which file out of the Unet training should we put in the 'network or config file' slot? Would it be one of the epoch/.pt files, or another one?
- For the new Unet to segment the image according to expectations, would it be better to add the ground truth labels in the 'layer reference' slot? Or should I keep this as none?

I hope you can help me out. Thank you very much for your time and help :)

Kind regards,

ArtGP-79.

**Comment by jni** [comment]
> napari stops responding and crashes consistently

> Is there any specific reason for the software to crash like this after finishing the segmentation? Should I use a more powerful computer to see if this fixes the issue or is this a file input error?

Did you get any kind of error log? It's hard to know otherwise what is causing the crash and how it might be fixed. I don't think it's a computer power issue, more likely a bug in our code 😅 but we need more info to diagnose it.

> Which file out of the Unet training should we put in the 'network or config file' slot? Would it be one of the epoch/.pt files, or another one?

I think this is the .pt file indeed.

> For the new Unet to segment the image according to expectations, would it be better to add the ground truth labels in the 'layer reference' slot? Or should I keep this as none?

tbh I don't remember what this is for. 😅 @AbigailMcGovern?

**Comment by ArtGP-79** [comment]
Thank you for your fast reply!

1) Unfortunately, I don't get any error logs or any indications of what's going on because when the segmentation finishes, Napari goes 'white' and the windows message "program does not respond" appears on screen. I have tried waiting for Napari to respond, but have not managed to get it to work, so I always have to kill it with the task manager. 

I also don't get any message on the VSCode console (which is where I run Napari from). It simply freezes.

It is weird that the software just crashes like that, maybe it would be helpful if you try to do the same thing with my data? See if you also run into this problem...

2) Ok great, then on this part there's no problem, just a quick question. I set the retraining to run for 4 epochs and what I get back are .pt files named:
- my-unet
- my-unet_epoch-0
- ...1
- ...2
- ...3

I suppose the epoch 3 .pt file is the one with the most accurate version of the new Unet, but is this the case? or would it be the "my-unet"?

3) No problem, looking forward to @AbigailMcGovern 's reply :)

Thank you very much for your time and help.

Kind regards,

ArtGP-79.

**Comment by jni** [comment]
> what I get back are .pt files named:

the latest one. I would check the "last modified" or "created" timestamps on the files and pick the latest. 

> maybe it would be helpful if you try to do the same thing with my data? 

Sure, if you can subset things to something small enough to share but that still causes the error, and share also the .pt file, I can try to run it. But as a warning the next two weeks are really busy with conferences for me so I may not be able to do it until the end of the month.

**Comment by ArtGP-79** [comment]
Hi,

I just sent you the image I am working with, the .pt files (latest ones), and the ground truth with which I trained the new Unet so you can have a look for yourself.

Sorry to bother while you are so busy, hope everything goes well at the conferences.

Kind regards,

ArtGP-79.
